Feature/main page

Ktechen opened the pull request:
## Branch

Der Branch fügt ein Login und Registrierung Seite hinzu. In den beiden Seite werden die Eingabe des Users per Html/Javascript validiert und mit einem Regex überprüft. Zum Beispiel für Passwörter wird es, nach Length, [a-z], [0-9], [A-Z] und special chars geprüft. Das Password hat einen extra "Show" Switch, um den Typen von "password" auf "text" zusetzen. Dadurch kann die Benutzerfreundlichkeit vereinfacht werden. Auf der Registrierung Seite werden die Password noch zusätzlich auf Gleichheit überprüft. Die Registrierung Seite hat eine interaktiv Visualisierung mit Roten Icons (-) und Grünen Icons (+) für weitere Benutzerfreundlichkeit. 

### Style

Die beiden Seite haben einen eigenen Style und sind für Mobile geräte angepasst.

## Heroku:
 https://dripbox.herokuapp.com/

## Image Source:
Background-image-Login: https://pixabay.com/de/photos/schreibtisch-computer-bildschirme-1245714/
Image-Login: https://c.tenor.com/pBrzvwLzbwoAAAAi/hacking-hack.gif
Icons-Registration: Diese Icons sind Teil der Bootstrap library

#### Bei Fragen gerne ein Kommentar hinterlassen. 
(Gruppe 1 - s0573111)

Ktechen committed:
Add Login screen

Ktechen committed:
Add Image to Login Screen
Add Controller Model and View
Add javaScript

Ktechen committed:
Add Staticpages controller

Ktechen committed:
Static home helper

Ktechen committed:
Add Image to MainPage

Ktechen committed:
Add Javascript Regex validation for password

Ktechen committed:
Add Javascript Regex validation for password and Add Show Password btn

Ktechen committed:
Add func for ShowPassword

Ktechen committed:
Add Drip style

Ktechen committed:
Add Drip style

Ktechen committed:
Add background Image

Ktechen committed:
Add mobile view and Pc view

Ktechen committed:
Fix Style bug and added Registration site

Ktechen committed:
Fix Style bug and added Registration site

Ktechen committed:
Add style to registrations.scss

Ktechen committed:
Add new style to registrations.scss

Ktechen committed:
Add Js refectoring

Ktechen committed:
Add Js refectoring

Ktechen committed:
fix pg

Commit:
rewritten the database.yml for pg usage

postgresql docker setup and run commands added:
setup_docker.sh (to install docker and setup for psql)
run_local_docker.sh (run the local docker instance)

Ktechen committed:
test remove @Import "Bootstrap"

Ktechen committed:
Tests is background image

Ktechen committed:
Tests is background image

Ktechen committed:
Tests is background image

Ktechen committed:
Tests is background image

Ktechen committed:
Tests is background image

Ktechen committed:
Tests is background image

Ktechen committed:
Tests is background image

Ktechen committed:
Tests is background image

Ktechen committed:
Background img fixed

Ktechen committed:
gg ez heroku

Ktechen committed:
Add Js events for password validation

Ktechen committed:
remove gem angular and refactor addEvent from password validation

Ktechen committed:
refactor isEqual methode

Ktechen committed:
Add show btn Password

Ktechen committed:
remove first and username

Ktechen committed:
Add partials

Ktechen committed:
Add gif animation

Ktechen committed:
Add routes

Ktechen committed:
Add route for main page

Ktechen committed:
add more routes for home and reg page

Ktechen committed:
Final commit for easy part

Ktechen committed:
Add Password isEqual validation

viktorianer commented on Gemfile:
```diff
@@ -91,4 +91,4 @@ group :test do
 
 # Windows does not include zoneinfo files, so bundle the tzinfo-data gem
-gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
+gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
```
Bitte füge die `EditorConfig` wie vorgeschlagen zum Projekt hinzu und nutze diese.

Ktechen replied:
Ich bin von Windows to Linux gewechsel und dann das entfernt.

viktorianer replied:
`EditorConfig` brauchst du überall. Die gehört ins Repo.

Ktechen replied:
Die EditorConfig ist nicht default in Rails bei aber ich kann es hinzufügen.

viktorianer replied:
Das ist richtig. Hat auch nichts mit Rails zu tun, sondern mit unseren Qualitätsanforderungen an Projekt! Schau dazu die entsprechenden Folien an.

viktorianer commented on app/assets/stylesheets/application.css:
```diff
@@ -10,6 +10,6 @@
  * It is generally better to create a new file per style scope.
  *
- *= require_tree .
+ *= require_tree 
```
Das dürfte so nicht mehr funktionieren. Dieser Punkt ist sehr wichtig. Warum hast du diesen überhaupt entfernt?

Ktechen replied:
Fixed

viktorianer commented on README.md:
```diff
@@ -7,20 +7,19 @@ created with "rails new" on an Arch Linux System
 
 * Database initialization
-    - Postgres
+    - not yet integrated
```
Beschränkt bitte die Änderung am Code nur darauf, was das Feature benötigt!

Ktechen replied:
ja stimmt

viktorianer commented on app/assets/stylesheets/home_pages.scss:
```diff
@@ -0,0 +1,87 @@
+@import "bootstrap";
+
+#Home_html {
+  background-image: url("https://raw.githubusercontent.com/Ktechen/Dripbox/feature/MainPage/app/assets/images/desktop-1245714_1920.jpg");
```
Bitte die Asset-Pipeline von Rails benutzen. Direkte Links ins Web sind selten gut, höchsten zu einem CDN. Schaue bitte dazu https://guides.rubyonrails.org/asset_pipeline.html

viktorianer commented on app/controllers/home_pages_controller.rb:
```diff
@@ -0,0 +1,5 @@
+class HomePagesController < ApplicationController
+  def index
+
```
Unnötige, leere Zeile. Was sagt denn Rubocop dazu?

viktorianer commented on app/javascript/packs/PasswordHandler.js:
```diff
@@ -0,0 +1,80 @@
+export const MIN_LENGTH = 10;
```
Unter `../packs` liegen nur max. 2 Dateien, die den Gesamtpack für JS und CSS beschreiben und builden. 
Wenn du also neue JS Dateien hinzufügen willst, muss du das im eigenen Unterordner machen. Dann diese im `app/javascript/packs/application.js` einbinden, damit ein Pack gebaut wird.

Hier dazu mehr:
https://edgeguides.rubyonrails.org/webpacker.html#using-webpacker-for-javascript

Ktechen replied:
Der Javascript anteil wurde wie im Guide korrigiert.

viktorianer commented on app/javascript/packs/Registration.js:
```diff
@@ -0,0 +1,94 @@
+const min_height = 1001;
+
+/*
+function showPasswordParameter() {
+    document.addEventListener("click", () => {
+        const alert = document.getElementsByClassName("Reg_alerts");
+        for (let i = 0; i < alert.length; i++) {
+            alert[i].style.display = "inline-block";
+        }
+    });
+}
+*/
```
Wenn du diesen Code nicht mehr brauchst, entferne es.

Ktechen replied:
Alles klar, wurde entfernt.

viktorianer commented on app/javascript/packs/Registration.js:
```diff
@@ -0,0 +1,94 @@
+
+function events() {
+
+    password.addEventListener("keyup", () => {
+        validationHandler("lowercaseSVGPath", lowercaseRegex(password.value));
+        validationHandler("uppercaseSVGPath", uppercaseRegex(password.value));
+        validationHandler("numberSVGPath", numberRegex(password.value));
+        validationHandler("specialCharSVGPath", specialRegex(password.value));
+        validationHandler("lengthSVGPath", isLengthCorrect(password.value));
+        validationHandler("equalSVGPath", isEqual(password.value, passwordAgain.value));
+    });
```
Welchen Vorteil hast du hier, wenn du Validierung neu implementierst und nicht die Rails-Validierung nutzt? Das ist ein recht großer Aufwand, für Features die Rails bereits hat.
https://edgeguides.rubyonrails.org/active_record_validations.html

Ktechen replied:
Der Code ist keine richtige Validation sondern dient er der Visualisierung der Eingabe. Daher fehlt wie Sie gesagt haben die Active record validation von Rails.

viktorianer commented on app/javascript/packs/home.js:
```diff
@@ -0,0 +1,33 @@
+    const defaultSrc = image.src;
+
+    email.addEventListener("focusin", () => {
+        image.setAttribute("src", "https://c.tenor.com/pBrzvwLzbwoAAAAi/hacking-hack.gif");
```
Das hier geht an der Asset-Pipeline vorbei. Damit schaltest du ein sehr starkes Feature von Rails ganz ab.
https://guides.rubyonrails.org/asset_pipeline.html

viktorianer commented on app/views/home_pages/index.html.slim:
```diff
@@ -0,0 +1,34 @@
+html[id="Home_html" ]
+body[id="Home_Body"]
+.container
```
Diese Zeilen gehören hier nicht rein. Das gehört in die Layouts.
https://guides.rubyonrails.org/layouts_and_rendering.html

viktorianer commented on app/views/home_pages/index.html.slim:
```diff
@@ -0,0 +1,34 @@
+    = form_tag("/r", :method => "get") do
+      button.registieren.btn.btn-link[type="submit"]
+        | Registieren
+      p[class="copyright"]
+        .mt-5.mb-3
+          | © Dripbox
```
Ebenfalls ein Teil von Layout.

Ktechen replied:
Remove war veraltet

viktorianer commented on app/views/layouts/application.html.erb:
```diff
@@ -13,4 +13,5 @@
   <body>
     <%= yield %>
   </body>
+
```
Was passiert hier?

viktorianer commented on app/views/registrations/_alerts.html.erb:
```diff
@@ -0,0 +1,89 @@
+<div class="row">
```
Das kann Rails bereits, musst du nicht neu Implementieren. Oder welchen Vorteil hast du davon?
Schaue dazu:
https://guides.rubyonrails.org/active_record_validations.html

viktorianer commented on app/views/registrations/_userAgreement.html.erb:
```diff
@@ -0,0 +1,12 @@
+<div class="row">
```
Die Namen der Dateien werden in Rails nicht `_userAgreement.html.erb` sondern wie folgt geschrieben: `_user_agreement.html.erb`

Ktechen replied:
wurde geändert

viktorianer commented on app/views/registrations/_userAgreement.html.erb:
```diff
@@ -0,0 +1,12 @@
+<div class="row">
+  <div class="col">
+    <div class="form-check form-check-inline">
+      <input class="form-check-input" type="checkbox" id="useragreement" value="option1" required>
+      <label class="form-check-label" for="useragreement">Nutzungsbedingungen</label>
+    </div>
+    <div class="form-check form-check-inline">
+      <input class="form-check-input" type="checkbox" id="PrivacyPolicy" value="option1" required>
+      <label class="form-check-label" for="PrivacyPolicy">Datenschutzerklärung</label>
+    </div>
+  </div>
```
Du baust hier Features von Rails nach. Benutze hier das form-Object.

Schaue dazu:
https://guides.rubyonrails.org/form_helpers.html

viktorianer commented on app/views/registrations/_userInput.html.erb:
```diff
@@ -0,0 +1,24 @@
+  <div class="col">
+    <div class="row">
+      <div class="col">
+        <input type="text" class="form-control" placeholder="Username" aria-label="Username" required>
+      </div>
+    </div>
+    <div class="row">
+      <div class="col">
+        <input type="email" class="form-control" placeholder="email@a.de" aria-label="Email" required>
+      </div>
+    </div>
+    <div class="row">
+      <div class="col">
+        <input type="password" class="form-control" placeholder="Password" aria-label="Password" id="passwordId" required pattern="^(?=.*[0-9])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]{10,}$">
+      </div>
+    </div>
+    <div class="row">
+      <div class="col">
+        <input type="password" class="form-control" placeholder="Again Password" aria-label="Again Password" id="passwordAgainId" required pattern="^(?=.*[0-9])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]{10,}$">
+      </div>
```
Auch hier baust du Features von Rails nach.

viktorianer commented on app/views/registrations/index.html.erb:
```diff
@@ -0,0 +1,42 @@
+      </div>
+      <%= render "userInput" %>
+      <%= render "alerts" %>
+      <div class="form-check form-switch">
+        <input class="form-check-input" type="checkbox" id="Reg_ShowPassword">
+        <label class="form-check-label" for="Reg_ShowPassword">Show Password</label>
+      </div>
```
Nutze hier die Features von Rails mit Formhelper
https://guides.rubyonrails.org/form_helpers.html

viktorianer commented on app/views/registrations/index.html.erb:
```diff
@@ -0,0 +1,42 @@
+<% end %>
+</main>
+</body>
+<%= javascript_pack_tag "Registration" %>
```
Das gehört in Layout. Du hast praktisch mehrfach dein JS eingebunden und damit dein JS-Code mehrfach ausgeliefert.

Ktechen replied:
Ich habe jetzt die Javascript der Application.html.erb hinzugefügt

viktorianer commented on config/database.yml:
```diff
@@ -1,18 +1,8 @@
+#   Ensure the pg gem is defined in your Gemfile
+#   gem 'pg'
 #
 default: &default
   host: localhost
```
Funktioniert das? Da steht ja nicht, welche Adapter und DB benutzt wird. Dürfte gar nicht funktionieren. 
Ist aber auch nicht Teil von Feature.

notefox replied:
die Datei habe ich bearbeitet

den adapter nimmt er bei rails auch, wenn er unter dem host definiert wird
der zugehörige docker command zum testen, wäre:
```
docker run -it -d -p 5432:5432 -v pgdata:/var/lib/postgres/data -e POSTGRES_PASSWORD="$POSTGRESPROJECT_DATABASE_PASSWORD" --name postgres postgres:13.2-alpine
``` 
(der, den sie uns mal bereit gestellt haben)

viktorianer replied:
Das ist soweit klar, aber es fehlt folgende Angabe:

```
default: &default
  adapter: postgresql
  ...
  ```
  
Das funktioniert wahrscheinlich deshalb ohne diese Zeile, weil Rails von Postgres ausgeht oder schlicht mehrere Ports probiert. Da ActiveRecord ja mehrere DBs ansprechen kann. 

Ich würde dennoch diese Zeile hinzufügen, da die Config ohne diese Zeile nicht korrekt ist. Auch, wenn es funktioniert.

notefox replied:
meinen sie so?
```
default: &default
  host: localhost
  adapter: postgresql
  ...
```
ist es nicht schon so in der aktuellen database.yml ?
https://github.com/Ktechen/Dripbox/blob/feature/MainPage/config/database.yml

viktorianer replied:
Nein, du kannst doch hier sehen, dass dieser `adapter: postgresql` fehlt. :)

notefox replied:
ich bemerke es gerade... es wird tatsächlich nicht angezeigt :/ 
(tut mir leid fehler meiner seite, habe bisher nur den aktuellen feature branch angeschaut)

unter der "Files Changed" Übersicht fehlt aber noch um einiges mehr als nur der adapter...
eigentlich sollte database.yml noch "pool", "user" und "password" beinhalten, was aber im feature branch schon vor längerer zeit von mir gepusht wurde und zurzeit auch dort sind... :/ 

nun gut, falls das Problem nach dem mergen immernoch vorhanden ist, wird es direkt gepatched, 

das eigentlich richtige file existiert ja auch noch im main branch wie auch in einigen feature branches und auch im setup branch, kein großes Problem von unserer Seite

viktorianer replied:
Doch, das ist ein Problem! Wenn du es jetzt in main-Branch mergen wirst, ist das alles weg, genau wie hier.

Du musst es also vorher, also hier, beheben.

viktorianer commented on config/routes.rb:
```diff
@@ -1,3 +1,13 @@
+
+  get 'registrations', to: "registrations#index"
+  get 'registrations/index'
+  get "r", to: "registrations#index"
```
Was ist das für eine Route? Solch ein kurzer Name ist kein guter Style.

Ktechen replied:
Okay gut zu wissen und wurde jetzt geändert.

viktorianer commented on config/routes.rb:
```diff
@@ -1,3 +1,13 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+
+  get 'registrations', to: "registrations#index"
+  get 'registrations/index'
```
Doppelt?

Ktechen replied:
gefixed

viktorianer commented on app/views/home_pages/index.html.slim:
```diff
@@ -0,0 +1,34 @@
+    h1.Home_h1
+      b Dripbox
+      |  but better
+    form[method="POST" action="/test" ]
```
Wo ist das Model zum Formular?
Schaue dazu bitte:
https://guides.rubyonrails.org/active_record_basics.html
https://guides.rubyonrails.org/active_record_migrations.html
https://guides.rubyonrails.org/form_helpers.html

viktorianer reviewed (changes requested):
Ich habe jetzt nur das wichtigste markiert. Da ist noch viel mehr.

Jedenfalls ist das noch kein Rails. Da musst du noch nachlesen, wie das in Rails alles funktioniert. Die kommende Woche werden wir ein Teil auch in der Vorlesung anschauen.

Ktechen committed:
Move Javascript to application.html.erb
Add User Model

Ktechen committed:
Add User Model

Ktechen committed:
Add User has_secure_password

Ktechen committed:
Remove to short routes

viktorianer commented on db/migrate/20210527200903_create_users.rb:
```diff
@@ -0,0 +1,11 @@
+class CreateUsers < ActiveRecord::Migration[6.1]
+  def change
+    create_table :users do |t|
```
Wo ist das Model dazu bzw. ein Controller? Wozu ist diese Migration hier?

viktorianer replied:
Du muss die Migration auch noch ausführen, damit du die Schema-Datei neu schreibst.

Ktechen replied:
Achso ich dachte durch db:migrate erstellt sich das jeder selber

Ktechen replied:
Schema wurde geaddet

viktorianer commented on app/views/layouts/application.html.erb:
```diff
@@ -1,16 +1,21 @@
+
+<body>
+<%= yield %>
+<%= javascript_pack_tag "Registration" %>
+<%= javascript_pack_tag "home" %>
```
Das ist immer noch nicht richtig. Du musst die über die Zeile 10 laden. Was du aber machst, ist noch ein weiteres Pack zu bauen, was nicht so gut ist.

viktorianer commented on app/views/registrations/index.html.erb:
```diff
@@ -0,0 +1,41 @@
+    </div>
+  </div>
+</div>
+<%= form_with url: home_pages_index_url, method: :push do |form| %>
```
Das ist schon besser, aber immer noch nicht richtig. In Formular wird bei Rails immer ein Model benutzt.

Ktechen committed:
Remove index.html.slim and change to index.html.erb
Create javascript folder for home.js, PasswordHandler.js, Registration.js

Add User Controller
Add Function to User controller like new, create and destroy

Ktechen committed:
Change name of javascript folder to src
refactor index.html.erb

Ktechen committed:
remove move Image
(Javascript lädt nicht richtig)
add Image with asset pipeline
add Show Switch

viktorianer commented on app/views/home_pages/index.html.erb:
```diff
@@ -0,0 +1,40 @@
+      <h2>Please sign in</h2>
+    </div>
+    <div class="form-floating">
+      <%= form_with url: "/test", model: @user do |user| %>
```
Du benötigst `url` hier nicht, da die URL durch Rails mit Hilfe des Model-Objekts bestimmt werden soll.

viktorianer replied:
Allgemein wird hier nicht `user` als Objekt bereitgestellt, sondern das ist ein Form-Helper. Daher wird das Objekt nicht `user`, sondern `form` oder einfach `f` benannt.

viktorianer commented on app/views/home_pages/index.html.erb:
```diff
@@ -0,0 +1,40 @@
+        <%= user.password_field :password, class: "form-control", id: "floatingPassword", placeholder: "password" %>
+
+        <div class="form-check form-switch">
+          <input class="form-check-input" id="ShowSwitch" type="checkbox"><label class="form-check-label" for="ShowSwitch" id="Switchlabel">Show
```
Hier kannst du ebenfalls den Form-Helper nutzen um diese Chackbox zu erstellen.

viktorianer commented on app/views/layouts/application.html.erb:
```diff
@@ -1,16 +1,16 @@
-    <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
-    <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
-  </head>
+  <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
+  <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
```
Diese gehören in den Head.

Ktechen replied:
Das habe ich nie verschoben hä????? Ich habe in meinen letzten Commit nachgeschaut und dort habe ich nix geändert. 
Das verstehe ich ja überhaupt nicht ??????????

viktorianer replied:
Ahja, ich sehe jetzt auch, dass die Änderung ohne Sinn ist, aber die beiden Zeilen sind im Header. Also, so erstmal OK. 
Hier ist der Commit von dir dazu https://github.com/Ktechen/Dripbox/commit/4b544548332b7b5e0e724ea0313ca11cbbb421c1#diff-f43fe075643e681b2c01c2f853bb0c4299d135b47fcbd4da96890d521c49e3eb

Hat mit Feature nichts zu tun, ist nur Einrückung der Zeilen.

viktorianer commented on app/javascript/packs/application.js:
```diff
@@ -8,6 +8,7 @@ import Turbolinks from "turbolinks"
 import * as ActiveStorage from "@rails/activestorage"
 import "channels"
 
+
```
Unnötige Änderung.

viktorianer commented on app/javascript/src/Registration.js:
```diff
@@ -0,0 +1,87 @@
+const password = document.getElementById("passwordId");
+const passwordAgain = document.getElementById("passwordAgainId")
+
+const plus = "M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM8.5 4.5a.5.5 0 0 0-1 0v3h-3a.5.5 0 0 0 0 1h3v3a.5.5 0 0 0 1 0v-3h3a.5.5 0 0 0 0-1h-3v-3z";
+const minus = "M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h7a.5.5 0 0 0 0-1h-7z";
```
Was machen diese zwei Zeilen?
Sind das svg-Bilder? Das gehört nicht hier rein, sondern unter Assets.

Ktechen replied:
Sind von Bootstrap die svg-Bilder

viktorianer replied:
Habe ich mir gedacht. Das muss über die Assets geladen werden.

Ktechen committed:
fix Javascript functions for home.js and Registration.js

Ktechen committed:
Add schema db

Ktechen committed:
change url to asset_url

Ktechen committed:
Add RegHandle

Ktechen committed:
remove async

viktorianer reviewed (approved):
Also von meiner Seite sind noch sehr viele Punkte offen. Aber, das dauert mir bereits viel zu lange. Wir sollten nicht alle im Team aufhalten.

notefox commented on app/assets/stylesheets/home_pages.scss:
```diff
@@ -0,0 +1,87 @@
+@import "bootstrap";
+
+#Home_html {
+  background-image: url("desktop-1245714_1920.jpg");
```
"url()" kann probleme bei heroku aufweisen

ich würde empfehlen "asset-url()" zu verwenden

Ktechen replied:
Ja das Teste ich heute mal 👍

Ktechen committed:
Checkbox to Rails conversion

Ktechen committed:
renamed desktop-1245714_1920.jpg to BackgroundImage.jpg

Ktechen committed:
Change from url to asset_url

Ktechen committed:
Create Dir for Images

Ktechen committed:
Change path of images

Ktechen committed:
Add Rails Validations

Ktechen committed:
Add First Tests and Add Validation via Rails

Ktechen committed:
Change html to erb Rails

Ktechen committed:
Remove Js

notefox commented on app/assets/stylesheets/home_pages.scss:
```diff
@@ -0,0 +1,87 @@
+@import "bootstrap";
+
+#Home_html {
+  background-image: asset_url("app/assets/images/home_page/BackgroundImage.jpg");
```
schreibfehler:
"asset-url" nicht "asset_url"

Ktechen replied:
jetzt lädt das image

notefox commented on app/views/registrations/_user_agreement.html.erb:
```diff
@@ -0,0 +1,12 @@
+      <label class="form-check-label" for="PrivacyPolicy">Datenschutzerklärung</label>
+    </div>
+  </div>
+</div>g
```
wozu ist das "g" zum schluss hier?

Ktechen replied:
Ist schon weg haben das nur noch nicht gepusht :)

notefox replied:
Dann push mal ^^

Ktechen committed:
Add rails user form

Ktechen committed:
Remove duplicate html and body

Ktechen committed:
change asset_url to asset-url and run rubocop

Ktechen committed:
Merge branch 'main' of https://github.com/Ktechen/Dripbox into feature/MainPage

 Conflicts:
	Gemfile

Ktechen committed:
fix gem file

Ktechen committed:
Import home.js

Ktechen committed:
Add gif image with assets pipeline

Ktechen committed:
remove slim from Gemfile.lock

Ktechen committed:
add gif

Ktechen committed:
Remove old forms
Change text colour zu

viktorianer commented:
Guten Morgen. So langsam muss das mal gemerget werden. Andere aus der Gruppe warten ja schon echt lange.

Ktechen committed:
Fix js loaded file
